sdk/monitor/Microsoft.Azure.Management.Monitor/tests/BasicTests/DataCollectionRulesTests.cs

```diff
@@ -0,0 +1,550 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.Management.Monitor;
+using Microsoft.Azure.Management.Monitor.Models;
+using Microsoft.Rest;
+using Microsoft.Rest.Azure;
+using Monitor.Tests.Helpers;
+
+using Newtonsoft.Json;
+
+using Xunit;
+
+namespace Monitor.Tests.BasicTests
+{
+    public class DataCollectionRulesTests : TestBase
+    {
+        #region DCR Tests
+        [Fact]
+        [Trait("Category", "Mock")]
+        public void CreateDataCollectionRuleTest()
```

Comment:
Why are these tests mocked rather than using the test recording mechanism?

Reply:
In the instruction is mention "Add or update tests for the newly generated code" but it's not specific.
I took an example of another Test and adjust accordingly.
Should I remove this file?
For the recording mechanism, Is the json file created manually? or there's a tool/procedure for create it?

Reply:
The advantage of the recording mechanism is that it records responses during live tests with the service, and whenever the api-version changes, the recording is invalidated and you have to re-run the tests live.  In this case, maintainers of the test will have to compare the mocked responses with service responses each time there is a chance, which increases the chance of confusion and errors.

It is appropriate to use fully mocked responses like this to simulate responses that are difficult to get from the live service (for example, to siulate handling of 409 errors).  If you think this is such a case, where the service responses you are lookign for would be difficult to gt from a live test, then using this kind of mocking is appropriate.  I know that there are *some* monitor tests (particularly around alerts) that are very difficult to simulate in live tests.

Reply:
@markcowl Recorded Test Session added!